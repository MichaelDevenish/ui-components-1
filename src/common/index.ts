export { Props } from './types'
export { getColor } from './legacy'
