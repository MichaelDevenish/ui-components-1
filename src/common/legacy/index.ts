export { getColor } from './colors/getColor'
