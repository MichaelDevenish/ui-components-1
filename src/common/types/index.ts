/**
 * For Examples, see:
 * https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts
 * https://github.com/alm-tools/alm/blob/master/src/common/types.ts
 */
import { Size as _s, AllSizes } from './size'

namespace Props {
  export import Size = _s
}

export {
  Props,
  AllSizes
}
