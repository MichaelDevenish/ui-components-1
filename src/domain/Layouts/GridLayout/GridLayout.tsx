import React from 'react'
import { CSSTransition, TransitionGroup } from 'react-transition-group'

import { Props } from '../../../common'
import {
  CellAnimation,
  CellSize,
  GutterSize,
  HorizontalAlignment,
  IStyledCellProps,
  IStyledCellSizes,
  StyledCell,
  StyledGridLayout,
  VerticalAlignment
} from './style'

interface ICellOffsetDefinition {
  min?: number
  tablet?: number
  desktop?: number
  bigDesktop?: number
}

interface IGutterSizeDefinition {
  min?: GutterSize
  tablet?: GutterSize
  desktop?: GutterSize
  bigDesktop?: GutterSize
}

interface IHorizontalAlignmentDefinition {
  min?: HorizontalAlignment
  tablet?: HorizontalAlignment
  desktop?: HorizontalAlignment
  bigDesktop?: HorizontalAlignment
}

interface IVerticalAlignmentDefinition {
  min?: VerticalAlignment
  tablet?: VerticalAlignment
  desktop?: VerticalAlignment
  bigDesktop?: VerticalAlignment
}

interface IGridLayoutCell {
  /**
   * Key for the cell. This is important for animations, as cells with the same key will be kept unanimated
   * when transitioning. Defaults to the index of the cell.
   */
  key?: string | number
  /** The content to place within the cell */
  content?: JSX.Element | string | null
  /** The size this cell takes up within the grid */
  size?: IStyledCellSizes | CellSize
  /** The cell offset from the edge of the grid */
  offset?: ICellOffsetDefinition | number
  /** Animation style for adding/removing the cell (overrides the grid style) */
  animationStyle?: CellAnimation
  /** Component context */
  componentContext?: string
}

interface IGridLayoutProps {
  children: React.ReactElement<IStyledCellProps>
  /** The cells to place within the grid */
  cells?: IGridLayoutCell[]
  /** The horizontal alignment of the items within the grid */
  horizontalAlignment?: IHorizontalAlignmentDefinition | HorizontalAlignment
  /** The vertical alignment of the items within the grid */
  verticalAlignment?: IVerticalAlignmentDefinition | VerticalAlignment
  /** Adds gutters between cells as margin in the x direction */
  gutterMarginX?: IGutterSizeDefinition | GutterSize
  /** Adds gutters between cells as margin in the y direction */
  gutterMarginY?: IGutterSizeDefinition | GutterSize
  /** Adds gutters between cells as padding in the x direction */
  gutterPaddingX?: IGutterSizeDefinition | GutterSize
  /** Adds gutters between cells as padding in the y direction */
  gutterPaddingY?: IGutterSizeDefinition | GutterSize
  /** Determines the amount of columns within the grid */
  gridColumns?: number
  /** Animation style for adding/removing cells to the grid */
  animationStyle?: CellAnimation
  /** Component context */
  componentContext?: string
}

export class GridLayout extends React.PureComponent<IGridLayoutProps, never> {
  public static Cell = StyledCell
  public static HorizontalAlignment = HorizontalAlignment
  public static VerticalAlignment = VerticalAlignment

  public static defaultProps: Partial<IGridLayoutProps> = {
    gridColumns: 12,
    horizontalAlignment: HorizontalAlignment.Left,
    verticalAlignment: VerticalAlignment.Stretch,
    gutterMarginX: 'none',
    gutterMarginY: 'none',
    gutterPaddingX: 'none',
    gutterPaddingY: 'none'
  }

  public render (): JSX.Element {
    const {
      gridColumns,
      gutterMarginX,
      gutterMarginY,
      gutterPaddingX,
      gutterPaddingY,
      horizontalAlignment,
      verticalAlignment,
      componentContext,
      cells,
      children,
      animationStyle
    } = this.props

    let renderChildren: Array<React.ReactElement<IStyledCellProps>> = [<></>]

    if (cells) {
      renderChildren = cells.map(this.getCellForDefinition)
    } else if (children) {
      renderChildren = React.Children.map<
        React.ReactElement,
        React.ReactElement<IStyledCellProps>
      >(children, (child, index) => {
        const { size, offset } = child.props
        return (
          <CSSTransition
            key={index}
            classNames='grid-layout-cell-animation'
            timeout={200}
            data-component-type={Props.ComponentType.GridLayoutCell}
            data-component-context={child.props.componentContext}
          >
            {React.cloneElement(child, {
              size: size || 'auto',
              offset: offset || 0,
              gridColumns,
              animationStyle,
              gutterMarginX,
              gutterMarginY,
              gutterPaddingX,
              gutterPaddingY
            })}
          </CSSTransition>
        )
      })
    }

    return (
      <StyledGridLayout
        horizontalAlignment={horizontalAlignment!}
        verticalAlignment={verticalAlignment!}
        gutterMarginX={gutterMarginX!}
        gutterMarginY={gutterMarginY!}
        data-component-type={Props.ComponentType.GridLayout}
        data-component-context={componentContext}
      >
        <TransitionGroup component={null}>{renderChildren}</TransitionGroup>
      </StyledGridLayout>
    )
  }

  private getCellForDefinition = (
    cell: IGridLayoutCell,
    index: number
  ): JSX.Element => {
    const {
      gridColumns,
      gutterMarginX,
      gutterMarginY,
      gutterPaddingX,
      gutterPaddingY,
      animationStyle: gridAnimationStyle
    } = this.props
    const {
      key,
      content,
      size,
      offset,
      componentContext,
      animationStyle: cellAnimationStyle
    } = cell

    return (
      <CSSTransition
        key={key !== undefined ? key : index}
        classNames='grid-layout-cell-animation'
        timeout={200}
      >
        <StyledCell
          size={size || 'auto'}
          offset={offset || 0}
          animationStyle={cellAnimationStyle || gridAnimationStyle || 'none'}
          gridColumns={gridColumns!}
          gutterMarginX={gutterMarginX!}
          gutterMarginY={gutterMarginY!}
          gutterPaddingX={gutterPaddingX!}
          gutterPaddingY={gutterPaddingY!}
          data-component-type={Props.ComponentType.GridLayoutCell}
          data-component-context={componentContext}
        >
          {content}
        </StyledCell>
      </CSSTransition>
    )
  }
}

export {
  IGridLayoutCell
}
