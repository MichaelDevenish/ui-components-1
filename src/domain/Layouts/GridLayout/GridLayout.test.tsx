import { mount } from 'enzyme'
import React from 'react'

import { Variables } from '../../../common'
import { GridLayout } from './GridLayout'

describe('<GridLayout />', () => {
  it(`should render a GridLayout`, () => {
    const wrapper = mount(
      <GridLayout
        margins={{
          top: Variables.Spacing.s2XSmall,
          left: Variables.Spacing.s2XSmall,
          right: Variables.Spacing.s2XSmall,
          bottom: Variables.Spacing.s2XSmall
        }}
        horizontalAlignment={GridLayout.HorizontalAlignment.Center}
        verticalAlignment={GridLayout.VerticalAlignment.Stretch}
        gutterMarginX={Variables.Spacing.s2XSmall}
        gutterPaddingY={{ min: Variables.Layout.l2XSmall, desktop: 'none' }}
        cells={[
          {
            size: 10,
            content: <div>1</div>
          },
          {
            size: 2,
            content: <div>2</div>,
            displayType: 'flex',
            flexHorizontalAlignment: { desktop: GridLayout.HorizontalAlignment.Right, min: GridLayout.HorizontalAlignment.Center }
          },
          {
            size: { desktop: 3, tablet: 4 },
            offset: { tablet: 2 },
            content: <div>3</div>
          },
          {
            size: { desktop: 6, tablet: 4 },
            content: <div>4</div>
          },
          {
            size: 'auto',
            offset: 3,
            content: <div>5</div>
          }
        ]}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
