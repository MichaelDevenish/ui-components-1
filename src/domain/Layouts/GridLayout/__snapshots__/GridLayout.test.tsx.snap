// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridLayout /> should render a GridLayout 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  min-height: 0;
  min-width: 0;
  width: 100%;
}

.c2 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  min-height: 0;
  min-width: 0;
  width: 100%;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  min-height: 0;
  min-width: 0;
  width: 100%;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  min-height: 0;
  min-width: 0;
  width: 100%;
}

.c5 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  min-height: 0;
  min-width: 0;
  width: 100%;
}

@media only screen and (max-width:639px) {
  .c0 {
    margin-left: -2px;
    margin-right: -2px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (min-width:640px) and (max-width:1033px) {
  .c0 {
    margin-left: -2px;
    margin-right: -2px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (min-width:1034px) and (max-width:1439px) {
  .c0 {
    margin-left: -2px;
    margin-right: -2px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (min-width:1440px) {
  .c0 {
    margin-left: -2px;
    margin-right: -2px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (max-width:639px) {
  .c1 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    width: calc(83.33333333333333% - 4px);
  }
}

@media only screen and (min-width:640px) and (max-width:1033px) {
  .c1 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    width: calc(83.33333333333333% - 4px);
  }
}

@media only screen and (min-width:1034px) and (max-width:1439px) {
  .c1 {
    margin-left: 2px;
    margin-right: 2px;
    width: calc(83.33333333333333% - 4px);
  }
}

@media only screen and (min-width:1440px) {
  .c1 {
    margin-left: 2px;
    margin-right: 2px;
    width: calc(83.33333333333333% - 4px);
  }
}

@media only screen and (max-width:639px) {
  .c2 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    width: calc(16.666666666666668% - 4px);
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media only screen and (min-width:640px) and (max-width:1033px) {
  .c2 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    width: calc(16.666666666666668% - 4px);
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media only screen and (min-width:1034px) and (max-width:1439px) {
  .c2 {
    margin-left: 2px;
    margin-right: 2px;
    width: calc(16.666666666666668% - 4px);
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media only screen and (min-width:1440px) {
  .c2 {
    margin-left: 2px;
    margin-right: 2px;
    width: calc(16.666666666666668% - 4px);
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media only screen and (max-width:639px) {
  .c3 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    -webkit-flex: 1 1 0%;
    -ms-flex: 1 1 0%;
    flex: 1 1 0%;
    width: auto;
  }
}

@media only screen and (min-width:640px) and (max-width:1033px) {
  .c3 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    width: calc(33.333333333333336% - 4px);
    margin-left: calc(16.666666666666668% + 2px);
  }
}

@media only screen and (min-width:1034px) and (max-width:1439px) {
  .c3 {
    margin-left: 2px;
    margin-right: 2px;
    width: calc(25% - 4px);
    margin-left: calc(16.666666666666668% + 2px);
  }
}

@media only screen and (min-width:1440px) {
  .c3 {
    margin-left: 2px;
    margin-right: 2px;
    width: calc(25% - 4px);
    margin-left: calc(16.666666666666668% + 2px);
  }
}

@media only screen and (max-width:639px) {
  .c4 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    -webkit-flex: 1 1 0%;
    -ms-flex: 1 1 0%;
    flex: 1 1 0%;
    width: auto;
  }
}

@media only screen and (min-width:640px) and (max-width:1033px) {
  .c4 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    width: calc(33.333333333333336% - 4px);
  }
}

@media only screen and (min-width:1034px) and (max-width:1439px) {
  .c4 {
    margin-left: 2px;
    margin-right: 2px;
    width: calc(50% - 4px);
  }
}

@media only screen and (min-width:1440px) {
  .c4 {
    margin-left: 2px;
    margin-right: 2px;
    width: calc(50% - 4px);
  }
}

@media only screen and (max-width:639px) {
  .c5 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    -webkit-flex: 1 1 0%;
    -ms-flex: 1 1 0%;
    flex: 1 1 0%;
    width: auto;
    margin-left: calc(25% + 2px);
  }
}

@media only screen and (min-width:640px) and (max-width:1033px) {
  .c5 {
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 2px;
    margin-right: 2px;
    -webkit-flex: 1 1 0%;
    -ms-flex: 1 1 0%;
    flex: 1 1 0%;
    width: auto;
    margin-left: calc(25% + 2px);
  }
}

@media only screen and (min-width:1034px) and (max-width:1439px) {
  .c5 {
    margin-left: 2px;
    margin-right: 2px;
    -webkit-flex: 1 1 0%;
    -ms-flex: 1 1 0%;
    flex: 1 1 0%;
    width: auto;
    margin-left: calc(25% + 2px);
  }
}

@media only screen and (min-width:1440px) {
  .c5 {
    margin-left: 2px;
    margin-right: 2px;
    -webkit-flex: 1 1 0%;
    -ms-flex: 1 1 0%;
    flex: 1 1 0%;
    width: auto;
    margin-left: calc(25% + 2px);
  }
}

<GridLayout
  cells={
    Array [
      Object {
        "content": <div>
          1
        </div>,
        "size": 10,
      },
      Object {
        "content": <div>
          2
        </div>,
        "displayType": "flex",
        "flexHorizontalAlignment": Object {
          "desktop": "right",
          "min": "center",
        },
        "size": 2,
      },
      Object {
        "content": <div>
          3
        </div>,
        "offset": Object {
          "tablet": 2,
        },
        "size": Object {
          "desktop": 3,
          "tablet": 4,
        },
      },
      Object {
        "content": <div>
          4
        </div>,
        "size": Object {
          "desktop": 6,
          "tablet": 4,
        },
      },
      Object {
        "content": <div>
          5
        </div>,
        "offset": 3,
        "size": "auto",
      },
    ]
  }
  gridColumns={12}
  gutterMarginX={4}
  gutterMarginY="none"
  gutterPaddingX="none"
  gutterPaddingY={
    Object {
      "desktop": "none",
      "min": 16,
    }
  }
  horizontalAlignment="center"
  verticalAlignment="stretch"
>
  <styled.div
    data-component-type="grid_layout"
    gutterMarginX={4}
    gutterMarginY="none"
    horizontalAlignment="center"
    verticalAlignment="stretch"
  >
    <div
      className="c0"
      data-component-type="grid_layout"
    >
      <TransitionGroup
        childFactory={[Function]}
        component={null}
      >
        <CSSTransition
          classNames="grid-layout-cell-animation"
          in={true}
          key=".$0"
          onExited={[Function]}
          timeout={200}
        >
          <Transition
            appear={false}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={200}
            unmountOnExit={false}
          >
            <styled.div
              animationStyle="none"
              data-component-type="grid_layout_cell"
              displayType="block"
              flexHorizontalAlignments="left"
              gridColumns={12}
              gutterMarginX={4}
              gutterMarginY="none"
              gutterPaddingX="none"
              gutterPaddingY={
                Object {
                  "desktop": "none",
                  "min": 16,
                }
              }
              offsets={0}
              sizes={10}
            >
              <div
                className="c1"
                data-component-type="grid_layout_cell"
                sizes={10}
              >
                <div>
                  1
                </div>
              </div>
            </styled.div>
          </Transition>
        </CSSTransition>
        <CSSTransition
          classNames="grid-layout-cell-animation"
          in={true}
          key=".$1"
          onExited={[Function]}
          timeout={200}
        >
          <Transition
            appear={false}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={200}
            unmountOnExit={false}
          >
            <styled.div
              animationStyle="none"
              data-component-type="grid_layout_cell"
              displayType="flex"
              flexHorizontalAlignments={
                Object {
                  "desktop": "right",
                  "min": "center",
                }
              }
              gridColumns={12}
              gutterMarginX={4}
              gutterMarginY="none"
              gutterPaddingX="none"
              gutterPaddingY={
                Object {
                  "desktop": "none",
                  "min": 16,
                }
              }
              offsets={0}
              sizes={2}
            >
              <div
                className="c2"
                data-component-type="grid_layout_cell"
                sizes={2}
              >
                <div>
                  2
                </div>
              </div>
            </styled.div>
          </Transition>
        </CSSTransition>
        <CSSTransition
          classNames="grid-layout-cell-animation"
          in={true}
          key=".$2"
          onExited={[Function]}
          timeout={200}
        >
          <Transition
            appear={false}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={200}
            unmountOnExit={false}
          >
            <styled.div
              animationStyle="none"
              data-component-type="grid_layout_cell"
              displayType="block"
              flexHorizontalAlignments="left"
              gridColumns={12}
              gutterMarginX={4}
              gutterMarginY="none"
              gutterPaddingX="none"
              gutterPaddingY={
                Object {
                  "desktop": "none",
                  "min": 16,
                }
              }
              offsets={
                Object {
                  "tablet": 2,
                }
              }
              sizes={
                Object {
                  "desktop": 3,
                  "tablet": 4,
                }
              }
            >
              <div
                className="c3"
                data-component-type="grid_layout_cell"
                sizes={
                  Object {
                    "desktop": 3,
                    "tablet": 4,
                  }
                }
              >
                <div>
                  3
                </div>
              </div>
            </styled.div>
          </Transition>
        </CSSTransition>
        <CSSTransition
          classNames="grid-layout-cell-animation"
          in={true}
          key=".$3"
          onExited={[Function]}
          timeout={200}
        >
          <Transition
            appear={false}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={200}
            unmountOnExit={false}
          >
            <styled.div
              animationStyle="none"
              data-component-type="grid_layout_cell"
              displayType="block"
              flexHorizontalAlignments="left"
              gridColumns={12}
              gutterMarginX={4}
              gutterMarginY="none"
              gutterPaddingX="none"
              gutterPaddingY={
                Object {
                  "desktop": "none",
                  "min": 16,
                }
              }
              offsets={0}
              sizes={
                Object {
                  "desktop": 6,
                  "tablet": 4,
                }
              }
            >
              <div
                className="c4"
                data-component-type="grid_layout_cell"
                sizes={
                  Object {
                    "desktop": 6,
                    "tablet": 4,
                  }
                }
              >
                <div>
                  4
                </div>
              </div>
            </styled.div>
          </Transition>
        </CSSTransition>
        <CSSTransition
          classNames="grid-layout-cell-animation"
          in={true}
          key=".$4"
          onExited={[Function]}
          timeout={200}
        >
          <Transition
            appear={false}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={200}
            unmountOnExit={false}
          >
            <styled.div
              animationStyle="none"
              data-component-type="grid_layout_cell"
              displayType="block"
              flexHorizontalAlignments="left"
              gridColumns={12}
              gutterMarginX={4}
              gutterMarginY="none"
              gutterPaddingX="none"
              gutterPaddingY={
                Object {
                  "desktop": "none",
                  "min": 16,
                }
              }
              offsets={3}
              sizes="auto"
            >
              <div
                className="c5"
                data-component-type="grid_layout_cell"
                sizes="auto"
              >
                <div>
                  5
                </div>
              </div>
            </styled.div>
          </Transition>
        </CSSTransition>
      </TransitionGroup>
    </div>
  </styled.div>
</GridLayout>
`;
